File: .agent/skills/cnma-cap-backend/templates/java/srv/src/main/java/cnma/{{module_name}}/domain/__feature__/events/OnAfterDelete{{FeatureName}}Event.java
package cnma.{{module_name}}.domain.__feature__.events;

import com.sap.cds.services.RequestContext;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;

/**
 * OnAfterDelete{{FeatureName}}Event - AFTER DELETE event handler.
 * SRP: Handles post-deletion cleanup.
 */
@Slf4j
@Component
@RequiredArgsConstructor
public class OnAfterDelete{{FeatureName}}Event {

    private final RequestContext requestContext;

    public Object execute(Object _, Object req) {
        String userId = getUserId(req);
        log.info("AFTER DELETE {{FeatureName}} by user: {}", userId);

        // Post-deletion cleanup
        return null;
    }

    private String getUserId(Object req) {
        return "anonymous";
    }
}

File: .agent/skills/cnma-cap-backend/templates/node/srv/src/domain/__feature__/events/OnAfterDelete{{FeatureName}}Event.js
import cds from '@sap/cds';

/**
 * OnAfterDelete{{FeatureName}}Event - AFTER DELETE event handler.
 * SRP: Handles post-deletion cleanup.
 */
module.exports = {
    execute: async (_, req) => {
        try {
            const userId = req.user?.id || 'anonymous';
            const deletedId = req.params?.[0];

            cds.log('{{FeatureName}} deleted', {
                userId,
                deletedId,
                timestamp: new Date().toISOString()
            });
        } catch (error) {
            cds.log.error('Error in after delete {{FeatureName}}: ', error);
        }
    }
};